### ISSUE TITLE
tf.estimator训练中一些问题请教

### ISSUE THREAD

**Issue by wyl6** [Aifcce/FM-FFM#2]
大佬你好,我在tf.12环境中跑了下你的代码.

最后的**evaluate**的auc和accuracy结果和**训练后期显示的**auc和accuracy差别很大,请问这是为什么啊?

`
INFO:tensorflow:Not using Distribute Coordinator.
INFO:tensorflow:Running training and evaluation locally (non-distributed).
INFO:tensorflow:Start train and evaluate loop. The evaluate will happen after every checkpoint. Checkpoint frequency is determined based on RunConfig arguments: save_checkpoints_steps None or save_checkpoints_secs 600.
INFO:tensorflow:Calling model_fn.
Tensor("line/Add:0", shape=(?, 1), dtype=float32) Tensor("add_13:0", shape=(?, 1), dtype=float32)
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Create CheckpointSaverHook.
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Restoring parameters from ./checkpoint/model.ckpt-12537
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Saving checkpoints for 12537 into ./checkpoint/model.ckpt.
INFO:tensorflow:accuracy = 0.0, auc = 0.0, loss = 0.43045193
INFO:tensorflow:loss = 0.43045193, step = 12537
INFO:tensorflow:accuracy = 0.8125, auc = 0.8545455, loss = 0.34445637 (1.974 sec)
INFO:tensorflow:accuracy = 0.8125, auc = 0.8814916, loss = 0.40068352 (0.889 sec)
INFO:tensorflow:accuracy = 0.8229167, auc = 0.8818183, loss = 0.42168415 (0.918 sec)
INFO:tensorflow:accuracy = 0.81640625, auc = 0.8789329, loss = 0.48822016 (0.925 sec)
INFO:tensorflow:global_step/sec: 88.7519
INFO:tensorflow:accuracy = 0.815625, auc = 0.8699351, loss = 0.3078098 (0.930 sec)
INFO:tensorflow:loss = 0.3078098, step = 13037 (5.636 sec)
INFO:tensorflow:accuracy = 0.8255208, auc = 0.8798088, loss = 0.43122578 (0.929 sec)
INFO:tensorflow:accuracy = 0.81696427, auc = 0.8794843, loss = 0.33564216 (0.919 sec)
INFO:tensorflow:accuracy = 0.8222656, auc = 0.8831697, loss = 0.49410835 (0.922 sec)
INFO:tensorflow:accuracy = 0.8211806, auc = 0.8775306, loss = 0.28416485 (0.944 sec)
INFO:tensorflow:global_step/sec: 107.733
INFO:tensorflow:accuracy = 0.828125, auc = 0.8812226, loss = 0.47806147 (0.926 sec)
INFO:tensorflow:loss = 0.47806147, step = 13537 (4.641 sec)
INFO:tensorflow:accuracy = 0.81960225, auc = 0.877296, loss = 0.4075935 (0.921 sec)
INFO:tensorflow:accuracy = 0.8229167, auc = 0.87728906, loss = 0.40545577 (0.923 sec)
INFO:tensorflow:accuracy = 0.82091343, auc = 0.87879324, loss = 0.43576178 (0.916 sec)
INFO:tensorflow:Saving checkpoints for 13930 into ./checkpoint/model.ckpt.
INFO:tensorflow:Calling model_fn.
Tensor("line/Add:0", shape=(?, 1), dtype=float32) Tensor("add_13:0", shape=(?, 1), dtype=float32)
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Starting evaluation at 2019-07-30-09:28:06
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Restoring parameters from ./checkpoint/model.ckpt-13930
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Finished evaluation at 2019-07-30-09:28:13
INFO:tensorflow:Saving dict for global step 13930: accuracy = 0.8372615, auc = 0.8942388, global_step = 13930, loss = 0.37255484
INFO:tensorflow:Saving 'checkpoint_path' summary for global step 13930: ./checkpoint/model.ckpt-13930
INFO:tensorflow:Loss for final step: 0.59781605.
`

如上所示,训练过程后期accuracy是0.8209,auc是0.8787
而最后evaluation结果accuracy是0.8372,auc是0.8942
训练和验证时输入的是相同的数据,这些指标的值应该接近啊.

即使训练过程中的log_hook参数改为每步都显示,在最后一步时accuracy还是0.82,不应该和验证结果一样吗?

求大神解答.

**Comment by wyl6** [comment]
> 
更新下，已解决~

简单来说就是，tf.estimator接口输出的结果是划动平均值，而非当前步的真实值
比如，第7步显示的accuracy是前6步的accuracy的均值，不是第7步的训练结束后在数据集上获得的实时结果

详细可以看下我之前的一篇实验记录：https://blog.csdn.net/xxiaobaib/article/details/98659036
